samples/nrf9160/pdn/src/main.c

```diff
@@ -25,6 +29,65 @@ static const char * const event_str[] = {
 	[PDN_EVENT_IPV6_DOWN] = "IPv6 down",
 };
 
+static void ifaddrs_print(void)
+{
+	int err;
+
+	struct nrf_ifaddrs *ifaddrs;
+	struct nrf_ifaddrs *ifa;
+
+	char addr[NRF_INET6_ADDRSTRLEN] = {0};
+	char netmask[NRF_INET6_ADDRSTRLEN] = {0};
+	char broadaddr[NRF_INET6_ADDRSTRLEN] = {0};
+	char dstaddr[NRF_INET6_ADDRSTRLEN] = {0};
+
+	err = nrf_getifaddrs(&ifaddrs);
+	if (err) {
+		printk("nrf_getifaddrs() failed, err %d\n", err);
+		return;
+	}
+
+	printk("\nInterface addresses:\n");
+
+	ifa = ifaddrs;
+	while (ifa != NULL) {
+		switch (ifa->ifa_netmask->sa_family) {
+		case NRF_AF_INET:
+			nrf_inet_ntop(ifa->ifa_addr->sa_family,
+				      &SIN(ifa->ifa_addr)->sin_addr,
+				      addr, sizeof(addr));
+			break;
+		case NRF_AF_INET6:
+			nrf_inet_ntop(ifa->ifa_addr->sa_family,
+				      &SIN6(ifa->ifa_addr)->sin6_addr,
```

Comment:
I think this should be `->sin_addr.s6_addr`, as the sample output seems to have scope & other bytes (not a correct IPv6 address)